File: code/community/1301705198959333384/reduce-latency-deepgram-zoom.go
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/deepgram-devs/deepgram-go-sdk/deepgramapi"
)

func main() {
	// Use context for cancellation
	ctx := context.Background()

	// Get the Deepgram API Key from environment variable
	apiKey, exists := os.LookupEnv("DEEPGRAM_API_KEY")
	if !exists {
		panic("DEEPGRAM_API_KEY environment variable not set")
	}

	// Initialize Deepgram client
	client := deepgramapi.NewClient(apiKey)

	// Define the audio stream source, e.g., from Zoom
	audioSource := "your-zoom-websocket-url"

	// Live stream options
	options := deepgramapi.LiveOptions{
		Punctuate:      true,
		InterimResults: true,
	}

	// Set up a live transcribe session	err := client.LiveAudio(ctx, audioSource, options, func(result deepgramapi.TranscriptResult) {
		for _, alternative := range result.Results {
			if alternative.IsFinal {
				fmt.Println("Final Transcript:", alternative.Alternatives[0].Transcript)
			}
		}
	})

	if err != nil {
		fmt.Println("Error during transcription:", err)
	}

	// Follow the Deepgram Go SDK documentation for more options and configurations
}

File: code/community/1301705198959333384/reduce-latency-deepgram-zoom.py
import os
from deepgram import Deepgram
import asyncio

async def main():
    # Set your Deepgram API key using an environment variable
    deepgram_api_key = os.getenv('DEEPGRAM_API_KEY')

    if not deepgram_api_key:
        raise ValueError("DEEPGRAM_API_KEY environment variable not set")

    # Initialize Deepgram SDK
    dg_client = Deepgram(deepgram_api_key)

    # Define the audio source (e.g., from a Zoom call)
    audio_source = 'your-zoom-websocket-url'

    # Start listening with additional options for lower latency
    try:
        # Connect to the audio stream
        response = await dg_client.transcription.live({
            'url': audio_source,
            'punctuate': True,
            'interim_results': True
        })

        # Process received transcriptions
        async for transcription in response:
            print(f'Transcription: {transcription}')

    except Exception as e:
        print(f'Error: {e}')

# Run the main function
asyncio.run(main())

# Refer to the Deepgram Python SDK documentation for more details on customization and handling results.